django-backend/fecfiler/reports/views.py

```diff
@@ -177,6 +180,70 @@ def list(self, request, *args, **kwargs):
         serializer = self.get_serializer(queryset, many=True)
         return Response(serializer.data)
 
+    @action(
+        detail=False,
+        methods=["get"],
+        url_path="can_delete",
+    )
+    def can_delete(self, request):
+        report_id = (request.query_params.get("report_id")
+                     or request.data.get("report_id"))
+        report_id = report_id.replace("-", "")
+        with connection.cursor() as cursor:
```

Comment:
We should be able to write these queries with the Django framework.  Although maybe a reason to handwrite it would be to do it in one query.  You could probably do select on transactions, a join on the relation table for reports, and another join on the transaction table where the back_reference_id equals the row's transaction id.  they you filter by rows that were joined with the back_reference_id

Reply:
In addition to Todd's comments, since the delete check is only needed on the Manage Reports page, the query_set could only be modified to add the canDelete flag in an override of the list() DRF view. The SQL joins have been problematic so isolating additional ones where we can is preferred. Also, if performance becomes and issue, coding it so that only transactions who turn up in the result set due to pagination get the delete flag annotation could be a possibility. See this new ticket for how we're looking into making this sort of modification to the Contact Manager which is having issues with query performance stemming from its delete flag: https://github.com/fecgov/fecfile-web-api/issues/799

Reply:
Ok. Thank you both so much for helping me with this. 
I removed this can_delete end point and instead built it in the serializer using django queries.
I also updated the delete functionality for reports and transactions to be hard deletes.
Additionally, apparently django 4.2.11 is now required by circle for dependency testing to pass, so I updated that here and in [the mozilla project.](https://github.com/mozilla/mozilla-django-oidc/pull/534)